torch/csrc/autograd/python_variable.cpp

```diff
@@ -154,6 +155,22 @@ static bool check_has_torch_dispatch(PyObject *obj) {
   );
 }
 
+
+static PyObject* device_to_py_class_ [static_cast<size_t>(c10::DeviceType::COMPILE_TIME_MAX_DEVICE_TYPES)];
+
+void RegisterPythonTensorClass(std::string device, PyObject* python_tensor_class) {
+  c10::Device dev(device);
+  if (dev.type() == kCPU || dev.type() == kCUDA) {
+    // TODO: should I throw something else?
+    throw new std::runtime_error("Cannot override classes for kCPU or kCUDA");
+  }
+  device_to_py_class_[static_cast<size_t>(dev.type())] = python_tensor_class;
```

Comment:
It might be good to check that someone didn't register something previously or warn on it?